app/src/main/java/com/hillnerds/soundsparrow/ChannelSoundGeneration.java

```diff
@@ -0,0 +1,177 @@
+
+            midiSequenceArray.add(note_on);
+            midiSequenceArray.add(note_off);
+        }
+        return midiSequenceArray;
+    }
+
+    /**
+     * Genrates a starting code for a note. Of the form XXXX CCCC where XXXX is the code
+     * of the MidiSequnce (ie. 1001 Note on) and CCCC is the channel number (from 0 to 15)
+     * @param channelNumber - the number of tha channel for which the starting seqence will be generated
+     * @return - a total starting code value
+     */
+    public static int generateStartingCode(int channelNumber){
+        return 144 + channelNumber;
+    }
+
+    /**
+     * Generate a random pitch value for a channel based off of the scale assigned to it
+     * @param c - the channel for which a note is generated
+     * @return - a value between 0 and 127
+     */
+    public static int generatePitch(Channel c){
+        int randomGeneratedPitch = c.scaleValues[c.randomGenerator.nextInt((7 - 0) + 1)];
+
+        return randomGeneratedPitch;
+    }
+
+    /**
+     * Generate a velocity value
+     * @param random_generator - a random generator object for a Channel in which note is generated
+     * @return - an integer velocity value
+     */
+    public static int generateVelocity(Random random_generator){
+        //TODO: pass signal strength as well to change the velosity value
+        int rssi = -80;
+        int randomGeneratedVelocity = Math.abs(rssi);
+
+        return randomGeneratedVelocity;
+    }
+
+    /**
+     * Chooses a duration for the Note
+     * @param random_generator - random generator for current channel
+     * @return - a duartion in miliseconds
+     */
+    public static int generateNoteDuration(Random random_generator){
+        //TODO: make the duration fit within measures
+        int[] duration = new int[] {200, 400, 400};
+        int random_duration = duration[random_generator.nextInt((2-0) + 1) + 0];
+
+        return random_duration;
+    }
+
+    /**
+     * Called from within a channel constructor.
+     * Choose a note that will be a starting point for scale generation.
+     * @param instrument - the instrument which will rpovide the range of pitch requirements.
+     * @param randomGenerator - a random generator object
+     * @return - a value within a range of the chosen instrument (between 0 and 127)
+     */
+    public static  int chooseStartingNote (Instrument instrument, Random randomGenerator){
+        int startingNote = randomGenerator.nextInt((instrument.rangeMax -
+                instrument.rangeMin) + 1) + instrument.rangeMin;
+
+        return startingNote;
+    }
+
+    /**
+     * Called from within a channel constructor.
+     * Based on the emotion assigned to this channel chooses a major or minor scale and
+     * its respective step array.
+     * @param emotion - an emotion name
+     * @return - an array of steps for an appropriate scale
+     */
+    public static int[] chooseScaleStep(String emotion) {
```

Comment:
Maybe restructure this to be something more like

```
switch (emotion) {
    case "happy":
        return new int[] {steps...};
    case "sad":
        return new int[] {steps...};
etc
}
```
to avoid instantiating unused arrays every time